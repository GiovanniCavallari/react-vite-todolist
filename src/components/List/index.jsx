import React from 'react';
import P from 'prop-types';
import ListItem from '../ListItem';

import './styles.css';

function List({ items, onFinish, onRemove }) {
  return (
    items &&
    items.map((item, index) => (
      <ListItem
        key={index}
        text={item.text}
        done={item.done}
        onFinish={() => onFinish(index)}
        onRemove={() => onRemove(index)}
      />
    ))
  );
}

List.propTypes = {
  items: P.arrayOf(
    P.shape({
      text: P.string.isRequired,
      done: P.bool.isRequired,
    }),
  ).isRequired,
  onFinish: P.func.isRequired,
  onRemove: P.func.isRequired,
};

export default List;
