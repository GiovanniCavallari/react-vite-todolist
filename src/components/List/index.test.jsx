import { fireEvent, render } from '@testing-library/react';
import List from './index';

describe('Test List Component', () => {
  const callback = jest.fn();

  const items = [
    {
      text: 'Sample To Do',
      done: false,
    },
    {
      text: 'Sample Do To',
      done: true,
    },
  ];

  const component = (
    <List items={items} onFinish={callback} onRemove={callback} />
  );

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { asFragment } = render(component);
    expect(asFragment).toMatchSnapshot();
  });

  it('should click on callbacks', () => {
    const { asFragment, getAllByTestId } = render(component);
    const buttons = getAllByTestId('button');

    buttons.forEach((btn) => {
      fireEvent.click(btn);
    });

    expect(asFragment).toMatchSnapshot();
    expect(callback).toBeCalledTimes(4);
  });
});
