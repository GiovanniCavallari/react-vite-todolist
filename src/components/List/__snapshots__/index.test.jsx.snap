// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List Component should click on callbacks 1`] = `[Function]`;

exports[`Test List Component should render correctly 1`] = `[Function]`;
